pkg/datapath/loader/base.go

```diff
@@ -277,6 +277,35 @@ func (l *loader) reinitializeOverlay(ctx context.Context, tunnelConfig tunnel.Co
 	return nil
 }
 
+func (l *loader) reinitializeWireguard(ctx context.Context) (err error) {
+	// to-wireguard bpf is only used for rev-DNAT, which is only needed when NodePort, KPR, native routing and L7 proxy are enabled together
+	if !option.Config.EnableWireguard ||
+		!option.Config.EnableNodePort ||
+		!option.Config.EnableL7Proxy ||
+		option.Config.RoutingMode != option.RoutingModeNative ||
+		option.Config.KubeProxyReplacement != option.KubeProxyReplacementTrue {
+		return
+	}
+
+	link, err := netlink.LinkByName(wgTypes.IfaceName)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve link for interface %s: %w", wgTypes.IfaceName, err)
+	}
+
+	opts := []string{
+		fmt.Sprintf("-DSECLABEL=%d", identity.ReservedIdentityWorld),
+		fmt.Sprintf("-DTHIS_INTERFACE_MAC={.addr=%s}", mac.CArrayString(link.Attrs().HardwareAddr)),
+		fmt.Sprintf("-DCALLS_MAP=cilium_calls_wireguard_%d", identity.ReservedIdentityWorld),
+
+		"-DENABLE_NODEPORT=1",
+	}
```

Comment:
This should also come via the [node_config.h](https://github.com/cilium/cilium/blob/99adddceccdc120ad75e6e63c79c9929fa813a59/pkg/datapath/linux/config/config.go#L426), I believe.